src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -502,19 +502,33 @@ g.test('always_returns_device')
   .fn(async t => {
     const { compatibilityMode } = t.params;
     const gpu = getGPU(t.rec);
-    // MAINTENANCE_TODO: Remove this cast compatibilityMode is added.
-    const adapter = await gpu.requestAdapter({ compatibilityMode } as GPURequestAdapterOptions);
+    // MAINTENANCE_TODO: Remove compatibilityMode and the cast once compatibilityMode is no longer
+    // used (mainly in `setDefaultRequestAdapterOptions`).
+    const adapter = await gpu.requestAdapter({
+      compatibilityMode,
+      featureLevel: compatibilityMode ? 'compatibility' : 'core',
+    } as GPURequestAdapterOptions);
     if (adapter) {
+      const device = await t.requestDeviceTracked(adapter);
+      assert(device instanceof GPUDevice, 'requestDevice must return a device or throw');
+
       if (!compatibilityMode) {
-        // This check is to make sure something lower-level is not forcing compatibility mode
-        // MAINTENANCE_TODO: Remove this cast compatibilityMode is added.
+        // This check is to make sure something lower-level is not forcing compatibility mode.
+
+        // MAINTENANCE_TODO: Simplify this check (and typecast) once we standardize how to do this.
+        const adapterExtensions = adapter as unknown as {
+          isCompatibilityMode?: boolean;
+          featureLevel?: string;
+        };
         t.expect(
-          !(adapter as unknown as { isCompatibilityMode?: boolean }).isCompatibilityMode,
-          'must not be compatibility mode'
+          // Old version of Compat design.
+          !adapterExtensions.isCompatibilityMode &&
+            // Current version of Compat design, as of this writing.
+            adapterExtensions.featureLevel !== 'compatibility' &&
+            // An unlanded proposed change to the Compat design, but it doesn't hurt to just check.
```

Comment:
Shall we do this as a separate step as it comes out of no where? Or at least link to the draft PR?